analysis/topEFT/dataDrivenEstimation.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+from coffea import hist, processor
+from topcoffea.modules.YieldTools import YieldTools
+from topcoffea.modules.GetValuesFromJsons import get_lumi
+import cloudpickle
+
+class DataDrivenProducer: 
+    def __init__(self, inputHist, outputName, year, doDDFakes=True, doDDFlips=False):
+        self.yt=YieldTools()
+        if type(inputHist) == str: # we are plugging a pickle file
```

Comment:
I suggest adding an additional check for the file extension, rather than just relying on the variable being a string, e.g.:
```suggestion
        if isinstance(inputHist,str) and inputHist.endwith('.pkl'): # we are plugging a pickle file
```

Reply:
thanks for the suggestion, I've applied it